src/sample-executors/RelayOrderExecutor.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {Owned} from "solmate/src/auth/Owned.sol";
+import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
+import {SignedOrder} from "UniswapX/src/base/ReactorStructs.sol";
+import {CurrencyLibrary} from "UniswapX/src/lib/CurrencyLibrary.sol";
+import {Permit2Lib} from "permit2/src/libraries/Permit2Lib.sol";
+import {IRelayOrderReactor} from "../interfaces/IRelayOrderReactor.sol";
+
+/// @notice Sample executor for Relay orders
+contract RelayOrderExecutor is Owned {
+    using SafeTransferLib for ERC20;
+    using CurrencyLibrary for address;
+
+    /// @notice thrown if this contract is called by an address other than the whitelisted caller
+    error CallerNotWhitelisted();
+
+    event WhitelistedCallerChanged(address indexed newWhitelistedCaller);
+
+    address private whitelistedCaller;
+    IRelayOrderReactor private immutable reactor;
+
+    modifier onlyWhitelistedCaller() {
+        if (msg.sender != whitelistedCaller) {
+            revert CallerNotWhitelisted();
+        }
+        _;
+    }
+
+    constructor(address _whitelistedCaller, IRelayOrderReactor _reactor, address _owner) Owned(_owner) {
+        whitelistedCaller = _whitelistedCaller;
+        reactor = _reactor;
+    }
+
+    /// @notice Shortcut to execute a single relay order
+    /// @param order The order to execute
+    function execute(SignedOrder calldata order) external onlyWhitelistedCaller {
```

Comment:
I think that with multicall existing ON the reactor we really dont need these wrapper functions and this contract isn't really an executor anymore but a fee controller... no?